rubocop.yml

```diff
@@ -3,6 +3,22 @@ inherit_mode:
   - Exclude
   - Include
 
+<%
+  # Identify installed RuboCop plugins to configure
+  plugin_configs = {
+    "rubocop-graphql" => "rubocop.graphql.yml",
+    "rubocop-minitest" => "rubocop.minitest.yml",
+    "rubocop-performance" => "rubocop.performance.yml",
+    "rubocop-rails" => "rubocop.rails.yml",
+    "rubocop-rake" => "rubocop.graphql.yml",
+    "rubocop-rspec" => "rubocop.rspec.yml",
+  }.select { |plugin_name, _| Gem.loaded_specs.include?(plugin_name) }
+%>
+
+require: <%= plugin_configs.keys.to_json %>
+
+inherit_from: <%= plugin_configs.values.to_json %>
```

Comment:
This is where the magic happens. 🪄 ✨ 👮‍♂️

Are there reasons why this would be a bad idea?

### Pros

- Allows us to inject config without forcing the dependency on any plugins, or knowing anything fancy about the host app.
- Removes the need to separate gems (e.g. `rubocop-rails-shopify` or whatever).
- Removes the need for consumers to opt-into the configs themselves. They can simply follow the suggestions from Rubocop, and as soon as they add a plugin it will "magically" be configured.

### Cons

- Doesn't allow us to enforce the version of the plugins.
  - This might get tricky as cops are added/removed.
    - We would probably have to fallback to enforcing the version in ERB, which isn't ideal.
- Could there be cases where it's unsafe to enable these?
  - We could move the magic into a special file (e.g. `infer-rubocop.yml`), and keep this file simple, but that would mean consumers would have to know to switch.
  - Maybe it's sufficient to require consumers to simply override the config if it isn't compatible.

Reply:
I've tentatively moved this into `rubocop.infer.yml`, since this repo itself ran into inapplicable cops. Although that may be a special case, so we can always put it back. If we went with this separate file, then it would mean consumers could opt into the magic

```yaml
inherit_gem:
  rubocop-shopify: rubocop.infer.yml
```

or stick with the current minimal approach:

```yaml
inherit_gem
  rubocop-shopify: rubocop.yml
```

or take a custom approach

```yaml
inherit_gem
  rubocop-shopify:
    - rubocop.yml
    - rubocop.rails.yml
```